pos_stock_available_online/models/stock_notifier_pos_mixin.py

```diff
@@ -0,0 +1,54 @@
+from odoo import models
+
+
+class StockNotifierPosMixin(models.AbstractModel):
+    _name = "stock.notifier.pos.mixin"
+    _description = "Stock Notifier POS Mixin"
+
+    def _prepare_pos_message(self, warehouse=None):
+        """
+        Return prepared message to send to POS
+        """
+        self.ensure_one()
+        warehouse = warehouse or self.warehouse_id
+        return warehouse._prepare_vals_for_pos(self.product_id)
+
+    def _skip_notify_pos(self):
+        """
+        Skip notification to POS
+        """
+        return False
+
+    def _get_warehouses_to_notify(self):
+        self.ensure_one()
+        return self.warehouse_id
```

Comment:
Since the current model is a mixin and can be plugged into any model this field may be missing which will cause an error, possible solution is to either return the model object `return self.env["stock.warehouse"]` or throw an exception `raise NotImplementedError()`

In the models where this mixin is connected, implement a method that returns the necessary warehouses

Reply:
This mixin was intended just to be inherited by `stock.move` and `stock.quant`, but I modified it as you proposed to be more extendable